app/models/school.rb

```diff
@@ -73,9 +85,13 @@ def next_school_in_responsible_body_when_sorted_by_name_ascending
     responsible_body.next_school_sorted_ascending_by_name(self)
   end
 
-  def can_order_devices?(device_type = 'std_device')
+  def has_devices_available_to_order?(device_type = 'std_device')
+    available_devices_count(device_type).positive?
+  end
+
+  def available_devices_count(device_type = 'std_device')
     allocation = device_allocations.by_device_type(device_type).first
-    allocation&.cap.to_i > allocation&.devices_ordered.to_i
+    allocation&.cap.to_i - allocation&.devices_ordered.to_i
```

Comment:
I wonder whether this logic should be on the allocation and called from here...

Reply:
Yes, good call